### PR TITLE
[NEW] Option to read firebase service account from file  

### PR DESCRIPTION
**Pull request by abhinavkrin**
Set GOOGLE_CREDS_PATH to the path of the service account JSON file downloaded from the firebase console. The initAuth function would read projectId, clientEmail and privateKey from json file instead of FIREBASE_PROJECT_ID, FIREBASE_CLIENT_EMAIL, FIREBASE_PRIVATE_KEY environment vars.

### CODE REVIEW

**Comment by debdutdeb** [comment]
Forcing workflow run.

File: app/components/auth/firebase/lib/functions.js [hunk 1]
```diff
@@ -42,6 +44,20 @@ export const createEmptyAuthUser = () => {
...
+    clientEmail: process.env.FIREBASE_CLIENT_EMAIL || null,
+    // The private key must not be accessible on the client side.
+    privateKey: process.env.FIREBASE_PRIVATE_KEY || null,
+  }
+  if(typeof window === 'undefined' && process.env.GOOGLE_CREDS_PATH){
+    // firebase admin has to be initialized using service acccount json file.
+    const googleCredsFromFile = getGoogleCredsFromFile(process.env.GOOGLE_CREDS_PATH);
+    googleCreds.projectId = googleCredsFromFile.project_id;
+    googleCreds.privateKey = googleCredsFromFile.private_key;
+    googleCreds.clientEmail = googleCredsFromFile.client_email;
```
Comment (RonLek) [inline app/components/auth/firebase/lib/functions.js:58]: Can we construct the googleCreds JSON directly within `getGoogleCredsFromFile.js` and return the ready object?

File: app/components/auth/firebase/lib/getGoogleCredsFromFile.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import fs from 'fs';
+export const getGoogleCredsFromFile = (path) => {
+  if(typeof window === 'undefined'){
+    const cert = fs.readFileSync(path);
+    return JSON.parse(cert);
+  }
+  throw new Error("This function can only be called from client side");
+}
```
Comment (RonLek) [inline app/components/auth/firebase/lib/getGoogleCredsFromFile.js:8]: New line please.

File: app/next.config.js [hunk 3]
```diff
@@ -7,5 +7,14 @@ module.exports = {
...
+  webpack: (config, { isServer }) => {
+    // Fixes npm packages that depend on `fs` module
+    if (!isServer) {
+      config.resolve.fallback = {
+        fs: false
+      }
+    }
+    return config
   }
 }
```
Comment (RonLek) [inline app/next.config.js:20]: New line please.

**Review by RonLek — CHANGES_REQUESTED** [review]

File: app/components/auth/firebase/lib/functions.js (hunk 1 above)
Comment (abhinavkrin) [reply app/components/auth/firebase/lib/functions.js]: Pushed the changes.

**Review by RonLek — APPROVED** [review]
LGTM!

**Pull request merged**